sources/builtins/export.c

```diff
@@ -128,10 +130,14 @@ int		export(t_cmd *cmd, t_shell *shell)
 		return (show_error_message(E_EXPORT, C_RED, "", X_EXPORT));
 	while (i < cmd->arg_count)
 	{
-		if (is_valid_export_arg(cmd->args[i]))
-			add_arg_to_env(shell, cmd->args[i]);
-		else
-			show_error_message(E_EXPORT, C_RED, cmd->args[i], X_EXPORT);
+			if (is_valid_export_arg(cmd->args[i]))
```

Comment:
`is_valid_export_arg()` should handle all checks. `count_delimiters()` should not be here.

Reply:
fair point. Issue was that export without the `export_delimiter` would segfault. Bash behavior would be this:

export a -> does nothing
export a in our minishell -> segfaults.

The idea was that the function should `return(0)` when no delimiter is found (and not an error!), essentially not doing anything like bash. Maybe you know a better approach? :O

Reply:
edit: fixed

Reply:
On second note: I put the only characters check in `is_valid_export_key`. As bash doesn't necessarily do anything with it, we can just argue that we handle that as an error instead. holds up the same result, not exporting anything in the env!